### ISSUE TITLE
[bug][v2][linux] fails to run on Fedora 40 tauri appimage built on GitHub Actions

### ISSUE THREAD

**Issue by anatawa12** [tauri-apps/tauri#10749]
### Describe the bug

fails to run on Fedora 40 tauri appimage built on GitHub Actions.

in console output, those log are shown 

```
Cannot get default EGL display: EGL_SUCCESS
Cannot create EGL context: invalid display (last error: EGL_SUCCESS)
Violación de segmento (`core' generado)
```

Downstream issue: vrc-get/vrc-get#1405

### Reproduction

1. Build tauri app with github actions. I made [this](https://github.com/anatawa12/tauri-simple-app/releases/tag/0.1.0) for testing
2. fails to launch tauri app due to error above

### Full `tauri info` output

```text
[✔] Environment
    - OS: Mac OS 14.4.1 X64
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.81.0-beta.2 (08328a323 2024-07-25)
    ✔ cargo: 1.81.0-beta.2 (a2b58c3da 2024-07-16)
    ✔ rustup: 1.27.1 (54dd3d00f 2024-04-24)
    ✔ Rust toolchain: beta-aarch64-apple-darwin (default)
    - node: 22.6.0
    - pnpm: 8.15.4
    - npm: 10.8.2

[-] Packages
    - tauri [RUST]: 2.0.0-rc.6
    - tauri-build [RUST]: 2.0.0-rc.6
    - wry [RUST]: 0.42.0
    - tao [RUST]: 0.29.1
    - tauri-cli [RUST]: 2.0.0-rc.3
    - @tauri-apps/api : not installed!
    - @tauri-apps/cli [NPM]: 2.0.0-rc.7

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: web
```

**Comment by naman-crabnebula** [comment]
Hi!

It is working fine here on Fedora 40 ( in VM hosted on Arch Linux via VirtualBox )

![image](https://github.com/user-attachments/assets/5875bc6a-d719-4546-a681-f74a8bb92acf)

Are you using wayland by any chance?

**Comment by anatawa12** [comment]
https://github.com/vrc-get/vrc-get/issues/1405#issue-2482510583

> I'm using Fedora 40, running Wayland.

Original reporter is using Wayland

**Comment by FabianLars** [comment]
It works on my Fedora system (a real one, not a VM) too but the error somewhat sounds like something nvidia specific which i can't test.

Can you ask your users to run the app with the `WEBKIT_DISABLE_COMPOSITING_MODE=1` and/or `WEBKIT_DISABLE_DMABUF_RENDERER=1` env var(s) set? This typically helps with nvidia specific issues (though this specific error is new to me so no promises)

**Comment by debbie-drg** [comment]
Original reporter here! The system where I was having the issue does indeed have an nvidia GPU. I've just tried running the Tauri simple app on a laptop without an nvidia GPU also running Fedora 40 and the application indeed works. I'll try those env variables on my other system with the nvidia GPU and report back once I get home in a few hours.

**Comment by debbie-drg** [comment]
Running the application using the `WEBKIT_DISABLE_DMABUF_RENDERER=1` env variable works.

**Comment by FabianLars** [comment]
Okay that's good i guess. It also means that this issue isn't really in our hands though :/ (as long as we keep using webkitgtk).

**Comment by anatawa12** [comment]
Is there anything I can do for improving this problem?

**Comment by FabianLars** [comment]
You could use std::env::set_var to set said env var for every user of your app. Of course that reduces performance for those where the renderer does work fine.

Alternatively, we can only wait for new webkitgtk releases. They are actively working on the renderer (which at the same time is the reason why webkitgtk broke for so many people in the first place).

**Comment by anatawa12** [comment]
I thought "as long as we keep using webkitgtk" means it's not fixable due to architecture or something hard to fix in webkitgtk.
but if it might be fixed, I'll just just wait for newer release. Thank you!

**Comment by Azuli-Bspt** [comment]
The workaround doesn't seem to work for me - I'm on CachyOS, arch-based system, and my setup is FULL AMD. I saw mentions of hybrid gpus too, so let me say that my processor DOES have integrated graphics, but they aren't being used because I got a proper gpu. Also, my error is different even in wayland.

No matter how I launch the appimage, I always seem to get the same error on the terminal:
Could not create default EGL display: EGL_BAD_PARAMETER. Aborting...
